### PR TITLE
Adds Two Racial Variant Mercenaries + Merc Additions

### PR DESCRIPTION
**Pull request by Stutternov**
## About The Pull Request

**Tl;dr**, PR does the following:
- Allows Elves to play Mercenary; idk why they can't since Drow and Tieflings can.
- Adds in new 'Black Oak Guardian' mercenary type for Half-Elves and Elves only. Polearm build with decent medium armor.
- Adds in new 'Underdweller' mercenary type for Drow and Dwarfs. Axe-build with a unique leather miners helmet (with a light on it!) and a unique pickaxe-axe combination (Steel axe, very slightly worse but is also a pickaxe)
- Gave Steppesman dodge expert; I see no reason not to since, unlike Rogues, Bards, and Assassins, he can't even wear medium armor. (Why are combat pilgrims better than literal mercs..)

The 'Underdweller' gets a decent bit of starting gear though it's main seller is that unique axe and the miners helmet; the helmet has half the light range of a torch nearly but it is an infinite light source, encouraging you to get a lamptern or make-do with the free light your leather helmet provides. 

The "Black Oak Guardian' looks akin to what one may think of a fantasy elf to look like! Golden-plated medium armor and a red cloak, the helmet being a sprite I found a bit ago on Blackstone of the beaky-helmet without the visor on it. Really fits the look! They get a spear as their polearm (Was tempted to make this a great mace or halberd, but I think a regular spear is honestly a really good starting weapon anyway - open to change it though!) and some nick-nacks. Basically Grenzel without the heavy armor training, but in return they get some other bonuses since they can't use heavy armor.

## Why It's Good For The Game

Firstly, I'm not sure why elves can't be mercenaries given every other race can; even Aasimar. I asked Alcoholism and got told 'go for it' in terms of elves being mercs, so - here you go. I think it fits thematically and allows flexibility in elf players to not just play pilgrim to be mercenaries.

Secondly, I feel these classes 'round out' the niche devs wanted for mercs. Each class having a niche to it skill wise. Now there will be one merc class for each type of weapon. Sword, bow, crossbow, heavy armor one, and now also an axe and polearm build.

And finally, I think these may help flesh out the feeling of mercenaries being foreigners in the land and make the world feel a tad bigger. Fully willing to edit the names as wanted for the merc-types however desired by staff.

## Pre-Merge Checklist

- [X] You tested this on a local server.
- [X] This code did not runtime during testing.
- [X] You documented all of your changes.

### CODE REVIEW

**Comment by NPC1314** [comment]
What the PR actually does seems to be spawn naked elf men. Testmerge failed.

**Comment by SilvernKell** [comment]
You did a great job on this. Just need to fix the bug. But I've been waiting for elf merc and it's really cool you went with polearm for them, thank you!

**Comment by Stutternov** [comment]
Bug is fixed, there was a few unfound ones as well (i.e missing helmet state) that I fixed. Should be good to go now with a handful of extra merc changes that I added due to recent code changes.

**Comment by messagecypher** [comment]
spear sprite's fucked for some reason, always off center

**Comment by Stutternov** [comment]
Oddly I don't think that's anything changed by this PR, that's just the current spear sprite issue.

**Comment by NikolayByondPlayer** [comment]
Sounds interesting
